### DISCUSSION TITLE
Is raster-vision suitable for tracing DTM data?

### DISCUSSION THREAD

**Discussion by zdila** [azavea/raster-vision#1667] [General]
Hello.

I have DTM (digital terrain model) data as GeoTIFFs and would like to trace forest tracks. You can see shaded relief output of such DTM at https://www.freemap.sk/#map=16/49.149721/21.155044&layers=5 . I would like to obtain vectors of the tracks.

![image](https://user-images.githubusercontent.com/636095/212385395-097999ec-b6ce-4d9e-8795-660db93a2d70.png)

Currently our OSM community is tracing it manually (from shaded relief) but I thought it could be really boosted with the help of ML. Is raster-vision able to do it? If yes, then I will start learning it.

**Comment by AdeelH** [comment]
>I would like to obtain vectors of the tracks.

It should indeed be possible do this with Raster Vision. This can be posed as a semantic segmentation task where you segment the image into two classes: "background" and "track". 

Raster Vision is also able to output segmentation predictions as vector polygons, so you will get your detected tracks as long thin polygons which you can erode to LineStrings (in shapely terminology) if that is what you require. 

The training data to the model would be manually traced tracks. These can be provided as either rasters or vectors. If vectors, these can be either polygons or LineStrings.

We have [an existing example](https://github.com/azavea/raster-vision/blob/master/rastervision_pytorch_backend/rastervision/pytorch_backend/examples/semantic_segmentation/spacenet_vegas.py) for detecting roads that I think is very similar and might be a helpful reference.

Please feel free to ask more questions!
